{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Lamprey Transcriptome Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "Camille Scott [camille dot scott dot w @gmail.com] [@camille_codon]\n",
    "\n",
    "camillescott.github.io\n",
    "\n",
    "Lab for Genomics, Evolution, and Development\n",
    "Michigan State University\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## About"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook is the entry point for the [Petromyzon marinus](http://nas.er.usgs.gov/queries/FactSheet.aspx?speciesID=836) (sea lamprey) de novo transcriptome analysis. This entry notebook contains links for the others, and code to collect and format data for the other notebooks. It should be run before all other notebooks in order to generate the requisite data."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Contents"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. [Transcript Analysis Notebook](petmar-transcripts.ipynb)\n",
    "2. [Tissue Analysis Notebook](petmar-tissues.ipynb)\n",
    "3. [Protein Analysis Notebook](petmar-proteins.ipynb)\n",
    "4. [Taxonomic Analysis Notebook](petmar-taxonomy.ipynb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "** Using data resources found in ../resources.json\n",
      "** Using config found in ../config.json\n"
     ]
    }
   ],
   "source": [
    "from libs import *\n",
    "%run -i common.ipy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "u'../_work/'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wdir()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Databases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>access</th>\n",
       "      <th>condition</th>\n",
       "      <th>db_type</th>\n",
       "      <th>filename</th>\n",
       "      <th>flowcell</th>\n",
       "      <th>label</th>\n",
       "      <th>meta_type</th>\n",
       "      <th>paired</th>\n",
       "      <th>phred</th>\n",
       "      <th>q_type</th>\n",
       "      <th>strand</th>\n",
       "      <th>terms</th>\n",
       "      <th>tissue</th>\n",
       "      <th>url</th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Myxinidae_pep</th>\n",
       "      <td>remote_query</td>\n",
       "      <td>NaN</td>\n",
       "      <td>prot</td>\n",
       "      <td>Myx.pep.all.fa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta_database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>uniprot</td>\n",
       "      <td>NaN</td>\n",
       "      <td>taxonomy:7762</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>braFlo_pep</th>\n",
       "      <td>remote_query</td>\n",
       "      <td>NaN</td>\n",
       "      <td>prot</td>\n",
       "      <td>braFlo.pep.all.fa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta_database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>uniprot</td>\n",
       "      <td>NaN</td>\n",
       "      <td>organism:7739+AND+keyword:1185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>danRer_pep</th>\n",
       "      <td>remote_file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>prot</td>\n",
       "      <td>danRer.pep.fa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta_database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ftp://ftp.ensembl.org/pub/release-75/fasta/dan...</td>\n",
       "      <td>43153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>homSap_pep</th>\n",
       "      <td>remote_file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>prot</td>\n",
       "      <td>homSap.pep.fa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta_database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ftp://ftp.ensembl.org/pub/release-74/fasta/hom...</td>\n",
       "      <td>104763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>lamp10</th>\n",
       "      <td>remote_file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nucl</td>\n",
       "      <td>lamp10.fasta</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>assembly</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>http://athyra.ged.msu.edu/~camille/lamprey/lam...</td>\n",
       "      <td>715345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>musMus_pep</th>\n",
       "      <td>remote_file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>prot</td>\n",
       "      <td>musMus.pep.fa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta_database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ftp://ftp.ensembl.org/pub/release-75/fasta/mus...</td>\n",
       "      <td>52165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>petMar2</th>\n",
       "      <td>remote_file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nucl</td>\n",
       "      <td>petMar2.fa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta_database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ftp://ftp.ensembl.org/pub/release-75/fasta/pet...</td>\n",
       "      <td>25006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>petMar2_cdna</th>\n",
       "      <td>remote_file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nucl</td>\n",
       "      <td>petMar2.cdna.fa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta_database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ftp://ftp.ensembl.org/pub/release-75/fasta/pet...</td>\n",
       "      <td>11489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>petMar2_cds</th>\n",
       "      <td>remote_file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nucl</td>\n",
       "      <td>petMar2.cds.fa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta_database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ftp://ftp.ensembl.org/pub/release-75/fasta/pet...</td>\n",
       "      <td>11442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>petMar2_gtf</th>\n",
       "      <td>remote_file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>feature</td>\n",
       "      <td>petMar2.gtf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gtf_database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>http://ftp.ensembl.org/pub/release-75/gtf/petr...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>petMar2_ncrna</th>\n",
       "      <td>remote_file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nucl</td>\n",
       "      <td>petMar2.ncrna.fa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta_database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ftp://ftp.ensembl.org/pub/release-75/fasta/pet...</td>\n",
       "      <td>2652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>petMar2_pep</th>\n",
       "      <td>remote_file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>prot</td>\n",
       "      <td>petMar2.pep.fa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fasta_database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ftp://ftp.ensembl.org/pub/release-75/fasta/pet...</td>\n",
       "      <td>11442</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     access condition  db_type           filename flowcell  \\\n",
       "Myxinidae_pep  remote_query       NaN     prot     Myx.pep.all.fa      NaN   \n",
       "braFlo_pep     remote_query       NaN     prot  braFlo.pep.all.fa      NaN   \n",
       "danRer_pep      remote_file       NaN     prot      danRer.pep.fa      NaN   \n",
       "homSap_pep      remote_file       NaN     prot      homSap.pep.fa      NaN   \n",
       "lamp10          remote_file       NaN     nucl       lamp10.fasta      NaN   \n",
       "musMus_pep      remote_file       NaN     prot      musMus.pep.fa      NaN   \n",
       "petMar2         remote_file       NaN     nucl         petMar2.fa      NaN   \n",
       "petMar2_cdna    remote_file       NaN     nucl    petMar2.cdna.fa      NaN   \n",
       "petMar2_cds     remote_file       NaN     nucl     petMar2.cds.fa      NaN   \n",
       "petMar2_gtf     remote_file       NaN  feature        petMar2.gtf      NaN   \n",
       "petMar2_ncrna   remote_file       NaN     nucl   petMar2.ncrna.fa      NaN   \n",
       "petMar2_pep     remote_file       NaN     prot     petMar2.pep.fa      NaN   \n",
       "\n",
       "              label       meta_type paired phred   q_type strand  \\\n",
       "Myxinidae_pep   NaN  fasta_database    NaN   NaN  uniprot    NaN   \n",
       "braFlo_pep      NaN  fasta_database    NaN   NaN  uniprot    NaN   \n",
       "danRer_pep      NaN  fasta_database    NaN   NaN      NaN    NaN   \n",
       "homSap_pep      NaN  fasta_database    NaN   NaN      NaN    NaN   \n",
       "lamp10          NaN        assembly    NaN   NaN      NaN    NaN   \n",
       "musMus_pep      NaN  fasta_database    NaN   NaN      NaN    NaN   \n",
       "petMar2         NaN  fasta_database    NaN   NaN      NaN    NaN   \n",
       "petMar2_cdna    NaN  fasta_database    NaN   NaN      NaN    NaN   \n",
       "petMar2_cds     NaN  fasta_database    NaN   NaN      NaN    NaN   \n",
       "petMar2_gtf     NaN    gtf_database    NaN   NaN      NaN    NaN   \n",
       "petMar2_ncrna   NaN  fasta_database    NaN   NaN      NaN    NaN   \n",
       "petMar2_pep     NaN  fasta_database    NaN   NaN      NaN    NaN   \n",
       "\n",
       "                                        terms tissue  \\\n",
       "Myxinidae_pep                   taxonomy:7762    NaN   \n",
       "braFlo_pep     organism:7739+AND+keyword:1185    NaN   \n",
       "danRer_pep                                NaN    NaN   \n",
       "homSap_pep                                NaN    NaN   \n",
       "lamp10                                    NaN    NaN   \n",
       "musMus_pep                                NaN    NaN   \n",
       "petMar2                                   NaN    NaN   \n",
       "petMar2_cdna                              NaN    NaN   \n",
       "petMar2_cds                               NaN    NaN   \n",
       "petMar2_gtf                               NaN    NaN   \n",
       "petMar2_ncrna                             NaN    NaN   \n",
       "petMar2_pep                               NaN    NaN   \n",
       "\n",
       "                                                             url    size  \n",
       "Myxinidae_pep                                                NaN    1004  \n",
       "braFlo_pep                                                   NaN   28544  \n",
       "danRer_pep     ftp://ftp.ensembl.org/pub/release-75/fasta/dan...   43153  \n",
       "homSap_pep     ftp://ftp.ensembl.org/pub/release-74/fasta/hom...  104763  \n",
       "lamp10         http://athyra.ged.msu.edu/~camille/lamprey/lam...  715345  \n",
       "musMus_pep     ftp://ftp.ensembl.org/pub/release-75/fasta/mus...   52165  \n",
       "petMar2        ftp://ftp.ensembl.org/pub/release-75/fasta/pet...   25006  \n",
       "petMar2_cdna   ftp://ftp.ensembl.org/pub/release-75/fasta/pet...   11489  \n",
       "petMar2_cds    ftp://ftp.ensembl.org/pub/release-75/fasta/pet...   11442  \n",
       "petMar2_gtf    http://ftp.ensembl.org/pub/release-75/gtf/petr...       0  \n",
       "petMar2_ncrna  ftp://ftp.ensembl.org/pub/release-75/fasta/pet...    2652  \n",
       "petMar2_pep    ftp://ftp.ensembl.org/pub/release-75/fasta/pet...   11442  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resources_df[resources_df.meta_type != 'sample']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>access</th>\n",
       "      <th>condition</th>\n",
       "      <th>db_type</th>\n",
       "      <th>filename</th>\n",
       "      <th>flowcell</th>\n",
       "      <th>label</th>\n",
       "      <th>meta_type</th>\n",
       "      <th>paired</th>\n",
       "      <th>phred</th>\n",
       "      <th>q_type</th>\n",
       "      <th>strand</th>\n",
       "      <th>terms</th>\n",
       "      <th>tissue</th>\n",
       "      <th>url</th>\n",
       "      <th>tissue_c</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>614GMAAXX_1_1_pf</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg18</td>\n",
       "      <td>nucl</td>\n",
       "      <td>614GMAAXX_1_1_pf.qc.fq.gz</td>\n",
       "      <td>614GM</td>\n",
       "      <td>embryo_stg18_614GM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>embryo</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>(0.996493656495, 0.998646674437, 0.702575950062)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614GMAAXX_2_1_pf</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg20</td>\n",
       "      <td>nucl</td>\n",
       "      <td>614GMAAXX_2_1_pf.qc.fq.gz</td>\n",
       "      <td>614GM</td>\n",
       "      <td>embryo_stg20_614GM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>embryo</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>(0.996493656495, 0.998646674437, 0.702575950062)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614GMAAXX_3_1_pf</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg22a</td>\n",
       "      <td>nucl</td>\n",
       "      <td>614GMAAXX_3_1_pf.qc.fq.gz</td>\n",
       "      <td>614GM</td>\n",
       "      <td>embryo_stg22a_614GM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>embryo</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>(0.996493656495, 0.998646674437, 0.702575950062)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614GMAAXX_4_1_pf</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg22b</td>\n",
       "      <td>nucl</td>\n",
       "      <td>614GMAAXX_4_1_pf.qc.fq.gz</td>\n",
       "      <td>614GM</td>\n",
       "      <td>embryo_stg22b_614GM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>embryo</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>(0.996493656495, 0.998646674437, 0.702575950062)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614GMAAXX_6_1_pf</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg23</td>\n",
       "      <td>nucl</td>\n",
       "      <td>614GMAAXX_6_1_pf.qc.fq.gz</td>\n",
       "      <td>614GM</td>\n",
       "      <td>embryo_stg23_614GM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>embryo</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>(0.996493656495, 0.998646674437, 0.702575950062)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614GMAAXX_7_1_pf</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg24c1</td>\n",
       "      <td>nucl</td>\n",
       "      <td>614GMAAXX_7_1_pf.qc.fq.gz</td>\n",
       "      <td>614GM</td>\n",
       "      <td>embryo_stg24c1_614GM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>embryo</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>(0.996493656495, 0.998646674437, 0.702575950062)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614GMAAXX_8_1_pf</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg24c2</td>\n",
       "      <td>nucl</td>\n",
       "      <td>614GMAAXX_8_1_pf.qc.fq.gz</td>\n",
       "      <td>614GM</td>\n",
       "      <td>embryo_stg24c2_614GM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>embryo</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>(0.996493656495, 0.998646674437, 0.702575950062)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB057JABXX_s_7_pe</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval</td>\n",
       "      <td>nucl</td>\n",
       "      <td>AB057JABXX_s_7_pe.trim.fq.gz</td>\n",
       "      <td>AB057</td>\n",
       "      <td>brain_larval_AB057</td>\n",
       "      <td>sample</td>\n",
       "      <td>True</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>brain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>(0.552941203117, 0.827450990677, 0.780392169952)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AI</th>\n",
       "      <td>local_file</td>\n",
       "      <td>adult</td>\n",
       "      <td>nucl</td>\n",
       "      <td>AI.fq.gz</td>\n",
       "      <td>614GM</td>\n",
       "      <td>intest_adult_614GM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AK</th>\n",
       "      <td>local_file</td>\n",
       "      <td>adult</td>\n",
       "      <td>nucl</td>\n",
       "      <td>AK.fq.gz</td>\n",
       "      <td>614GM</td>\n",
       "      <td>kidney_adult_614GM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BB0582ABXX_s_3_pe</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval</td>\n",
       "      <td>nucl</td>\n",
       "      <td>BB0582ABXX_s_3_pe.trim.fq.gz</td>\n",
       "      <td>BB058</td>\n",
       "      <td>liver_larval_BB058</td>\n",
       "      <td>sample</td>\n",
       "      <td>True</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BB0582ABXX_s_4_pe</th>\n",
       "      <td>local_file</td>\n",
       "      <td>pars</td>\n",
       "      <td>nucl</td>\n",
       "      <td>BB0582ABXX_s_4_pe.trim.fq.gz</td>\n",
       "      <td>BB058</td>\n",
       "      <td>liver_pars_BB058</td>\n",
       "      <td>sample</td>\n",
       "      <td>True</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BB0582ABXX_s_5_pe</th>\n",
       "      <td>local_file</td>\n",
       "      <td>adult</td>\n",
       "      <td>nucl</td>\n",
       "      <td>BB0582ABXX_s_5_pe.trim.fq.gz</td>\n",
       "      <td>BB058</td>\n",
       "      <td>brain_adult_BB058</td>\n",
       "      <td>sample</td>\n",
       "      <td>True</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>brain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>(0.552941203117, 0.827450990677, 0.780392169952)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BB0582ABXX_s_6_pe</th>\n",
       "      <td>local_file</td>\n",
       "      <td>adult_1</td>\n",
       "      <td>nucl</td>\n",
       "      <td>BB0582ABXX_s_6_pe.trim.fq.gz</td>\n",
       "      <td>BB058</td>\n",
       "      <td>intestine_adult_1_BB058</td>\n",
       "      <td>sample</td>\n",
       "      <td>True</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intestine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>(0.703098825847, 0.869942337625, 0.411657060245)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BB0582ABXX_s_7_pe</th>\n",
       "      <td>local_file</td>\n",
       "      <td>adult_2</td>\n",
       "      <td>nucl</td>\n",
       "      <td>BB0582ABXX_s_7_pe.trim.fq.gz</td>\n",
       "      <td>BB058</td>\n",
       "      <td>intestine_adult_2_BB058</td>\n",
       "      <td>sample</td>\n",
       "      <td>True</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intestine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>(0.703098825847, 0.869942337625, 0.411657060245)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BB0582ABXX_s_8_pe</th>\n",
       "      <td>local_file</td>\n",
       "      <td>adult_3</td>\n",
       "      <td>nucl</td>\n",
       "      <td>BB0582ABXX_s_8_pe.trim.fq.gz</td>\n",
       "      <td>BB058</td>\n",
       "      <td>unknown_adult_3_BB058</td>\n",
       "      <td>sample</td>\n",
       "      <td>True</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FW1-Gill</th>\n",
       "      <td>local_file</td>\n",
       "      <td>freshw</td>\n",
       "      <td>nucl</td>\n",
       "      <td>FW1-Gill.fq.gz</td>\n",
       "      <td>614H7</td>\n",
       "      <td>gill_freshw_614H7</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gill</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>(0.978685122378, 0.507312601454, 0.456655135926)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FW1-Intestine</th>\n",
       "      <td>local_file</td>\n",
       "      <td>freshw</td>\n",
       "      <td>nucl</td>\n",
       "      <td>FW1-Intestine.fq.gz</td>\n",
       "      <td>614H7</td>\n",
       "      <td>intest_freshw_614H7</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FW1-Kidney</th>\n",
       "      <td>local_file</td>\n",
       "      <td>freshw</td>\n",
       "      <td>nucl</td>\n",
       "      <td>FW1-Kidney.fq.gz</td>\n",
       "      <td>614H7</td>\n",
       "      <td>kidney_freshw_614H7</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G81-a</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_a_tfm0</td>\n",
       "      <td>nucl</td>\n",
       "      <td>G81-a.fq.gz</td>\n",
       "      <td>614GT</td>\n",
       "      <td>gill_larval_a_tfm0_614GT</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gill</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>(0.978685122378, 0.507312601454, 0.456655135926)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G81-b</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_b_tfm0</td>\n",
       "      <td>nucl</td>\n",
       "      <td>G81-b.fq.gz</td>\n",
       "      <td>61HWC</td>\n",
       "      <td>gill_larval_b_tfm0_61HWC</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gill</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>(0.978685122378, 0.507312601454, 0.456655135926)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G82-a</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_a_tfm6</td>\n",
       "      <td>nucl</td>\n",
       "      <td>G82-a.fq.gz</td>\n",
       "      <td>614GT</td>\n",
       "      <td>gill_larval_a_tfm6_614GT</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gill</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>(0.978685122378, 0.507312601454, 0.456655135926)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G82-b</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_b_tfm6</td>\n",
       "      <td>nucl</td>\n",
       "      <td>G82-b.fq.gz</td>\n",
       "      <td>61HWC</td>\n",
       "      <td>gill_larval_b_tfm6_61HWC</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gill</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>(0.978685122378, 0.507312601454, 0.456655135926)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G83-a</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_a_tfm12</td>\n",
       "      <td>nucl</td>\n",
       "      <td>G83-a.fq.gz</td>\n",
       "      <td>614GT</td>\n",
       "      <td>gill_larval_a_tfm12_614GT</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gill</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>(0.978685122378, 0.507312601454, 0.456655135926)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G83-b</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_b_tfm12</td>\n",
       "      <td>nucl</td>\n",
       "      <td>G83-b.fq.gz</td>\n",
       "      <td>61HWC</td>\n",
       "      <td>gill_larval_b_tfm12_61HWC</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gill</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>(0.978685122378, 0.507312601454, 0.456655135926)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gill</th>\n",
       "      <td>local_file</td>\n",
       "      <td></td>\n",
       "      <td>nucl</td>\n",
       "      <td>Gill.fq.gz</td>\n",
       "      <td>42UL4</td>\n",
       "      <td>gill_42UL4</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gill</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>(0.978685122378, 0.507312601454, 0.456655135926)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>JV1_liver</th>\n",
       "      <td>local_file</td>\n",
       "      <td>juv</td>\n",
       "      <td>nucl</td>\n",
       "      <td>JV1_liver.fq.gz</td>\n",
       "      <td>42UKJ</td>\n",
       "      <td>intest_juv_42UKJ</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>JV_intest</th>\n",
       "      <td>local_file</td>\n",
       "      <td>juv_b</td>\n",
       "      <td>nucl</td>\n",
       "      <td>JV_intest.fq.gz</td>\n",
       "      <td>42UKJ</td>\n",
       "      <td>liver_juv_b_42UKJ</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>JVa_liver</th>\n",
       "      <td>local_file</td>\n",
       "      <td>juv_1</td>\n",
       "      <td>nucl</td>\n",
       "      <td>JVa_liver.fq.gz</td>\n",
       "      <td>42UKJ</td>\n",
       "      <td>liver_juv_1_42UKJ</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>JVb_liver</th>\n",
       "      <td>local_file</td>\n",
       "      <td>juv_a</td>\n",
       "      <td>nucl</td>\n",
       "      <td>JVb_liver.fq.gz</td>\n",
       "      <td>42UKJ</td>\n",
       "      <td>liver_juv_a_42UKJ</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Kidney</th>\n",
       "      <td>local_file</td>\n",
       "      <td></td>\n",
       "      <td>nucl</td>\n",
       "      <td>Kidney.fq.gz</td>\n",
       "      <td>42UL4</td>\n",
       "      <td>kidney_42UL4</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L81-a</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_a_tfm0</td>\n",
       "      <td>nucl</td>\n",
       "      <td>L81-a.fq.gz</td>\n",
       "      <td>614GT</td>\n",
       "      <td>liver_larval_a_tfm0_614GT</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L81-b</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_b_tfm0</td>\n",
       "      <td>nucl</td>\n",
       "      <td>L81-b.fq.gz</td>\n",
       "      <td>61HWC</td>\n",
       "      <td>liver_larval_b_tfm0_61HWC</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L82-a</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_a_tfm6</td>\n",
       "      <td>nucl</td>\n",
       "      <td>L82-a.fq.gz</td>\n",
       "      <td>614GT</td>\n",
       "      <td>liver_larval_a_tfm6_614GT</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L82-b</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_b_tfm6</td>\n",
       "      <td>nucl</td>\n",
       "      <td>L82-b.fq.gz</td>\n",
       "      <td>61HWC</td>\n",
       "      <td>liver_larval_b_tfm6_61HWC</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L83-1-a</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_a1_tfm12</td>\n",
       "      <td>nucl</td>\n",
       "      <td>L83-1-a.fq.gz</td>\n",
       "      <td>614GT</td>\n",
       "      <td>liver_larval_a1_tfm12_614GT</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L83-1-b</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_b1_tfm12</td>\n",
       "      <td>nucl</td>\n",
       "      <td>L83-1-b.fq.gz</td>\n",
       "      <td>61HWC</td>\n",
       "      <td>liver_larval_b1_tfm12_61HWC</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L83-a</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_a_tfm12</td>\n",
       "      <td>nucl</td>\n",
       "      <td>L83-a.fq.gz</td>\n",
       "      <td>614GT</td>\n",
       "      <td>liver_larval_a_tfm12_614GT</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L83-b</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval_b_tfm12</td>\n",
       "      <td>nucl</td>\n",
       "      <td>L83-b.fq.gz</td>\n",
       "      <td>61HWC</td>\n",
       "      <td>liver_larval_b_tfm12_61HWC</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LI</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval</td>\n",
       "      <td>nucl</td>\n",
       "      <td>LI.fq.gz</td>\n",
       "      <td>434JM</td>\n",
       "      <td>intest_larval_434JM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LK</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval</td>\n",
       "      <td>nucl</td>\n",
       "      <td>LK.fq.gz</td>\n",
       "      <td>434JM</td>\n",
       "      <td>kidney_larval_434JM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lips</th>\n",
       "      <td>local_file</td>\n",
       "      <td></td>\n",
       "      <td>nucl</td>\n",
       "      <td>Lips.fq.gz</td>\n",
       "      <td>42UL4</td>\n",
       "      <td>lips_42UL4</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lips</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>(0.848304507312, 0.842768175462, 0.848396779509)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M0_liver</th>\n",
       "      <td>local_file</td>\n",
       "      <td>larval</td>\n",
       "      <td>nucl</td>\n",
       "      <td>M0_liver.fq.gz</td>\n",
       "      <td>42UKJ</td>\n",
       "      <td>liver_larval_42UKJ</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M2_liver</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg2</td>\n",
       "      <td>nucl</td>\n",
       "      <td>M2_liver.fq.gz</td>\n",
       "      <td>42UKJ</td>\n",
       "      <td>liver_stg2_42UKJ</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M5_liver</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg5</td>\n",
       "      <td>nucl</td>\n",
       "      <td>M5_liver.fq.gz</td>\n",
       "      <td>42UKJ</td>\n",
       "      <td>liver_stg5_42UKJ</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>liver</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>(0.738239154395, 0.508542897421, 0.741668604402)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Monocytes</th>\n",
       "      <td>local_file</td>\n",
       "      <td></td>\n",
       "      <td>nucl</td>\n",
       "      <td>Monocytes.fq.gz</td>\n",
       "      <td>61504</td>\n",
       "      <td>monocytes_61504</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>monocytes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>(0.801568639278, 0.921630146924, 0.769903896603)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Muscle</th>\n",
       "      <td>local_file</td>\n",
       "      <td>m_sperm</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Muscle.fq.gz</td>\n",
       "      <td>42UL4</td>\n",
       "      <td>muscle_m_sperm_42UL4</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>muscle</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>(1.0, 0.929411768913, 0.435294121504)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neutrophils</th>\n",
       "      <td>local_file</td>\n",
       "      <td></td>\n",
       "      <td>nucl</td>\n",
       "      <td>Neutrophils.fq.gz</td>\n",
       "      <td>61504</td>\n",
       "      <td>neutrophils_61504</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>neutrophils</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>(0.552941203117, 0.827450990677, 0.780392169952)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OFHS</th>\n",
       "      <td>local_file</td>\n",
       "      <td>f_ovu</td>\n",
       "      <td>nucl</td>\n",
       "      <td>OFHS.fq.gz</td>\n",
       "      <td>61504</td>\n",
       "      <td>headskin_f_ovu_61504</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>headskin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>(0.507727826343, 0.694256073587, 0.822237611168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POF-eye</th>\n",
       "      <td>local_file</td>\n",
       "      <td>f_preovu</td>\n",
       "      <td>nucl</td>\n",
       "      <td>POF-eye.fq.gz</td>\n",
       "      <td>61504</td>\n",
       "      <td>eye_f_preovu_61504</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>eye</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>(0.749096516301, 0.733656301919, 0.852502892298)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POF-ts</th>\n",
       "      <td>local_file</td>\n",
       "      <td>f_preovu</td>\n",
       "      <td>nucl</td>\n",
       "      <td>POF-ts.fq.gz</td>\n",
       "      <td>61504</td>\n",
       "      <td>tailskin_f_preovu_61504</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tailskin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>(0.507727826343, 0.694256073587, 0.822237611168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PSMGill</th>\n",
       "      <td>local_file</td>\n",
       "      <td>m_presperm</td>\n",
       "      <td>nucl</td>\n",
       "      <td>PSMGill.fq.gz</td>\n",
       "      <td>42UL4</td>\n",
       "      <td>gill_m_presperm_42UL4</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gill</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>(0.978685122378, 0.507312601454, 0.456655135926)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rope</th>\n",
       "      <td>local_file</td>\n",
       "      <td>adult_m</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Rope.fq.gz</td>\n",
       "      <td>42UL4</td>\n",
       "      <td>rope_adult_m_42UL4</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rope</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13</td>\n",
       "      <td>(0.996493656495, 0.998646674437, 0.702575950062)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SMGill</th>\n",
       "      <td>local_file</td>\n",
       "      <td>m_sperm</td>\n",
       "      <td>nucl</td>\n",
       "      <td>SMGill.fq.gz</td>\n",
       "      <td>42UL4</td>\n",
       "      <td>gill_m_sperm_42UL4</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gill</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>(0.978685122378, 0.507312601454, 0.456655135926)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SMHS</th>\n",
       "      <td>local_file</td>\n",
       "      <td>m_sperm</td>\n",
       "      <td>nucl</td>\n",
       "      <td>SMHS.fq.gz</td>\n",
       "      <td>61504</td>\n",
       "      <td>headskin_m_sperm_61504</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>headskin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>(0.507727826343, 0.694256073587, 0.822237611168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SN</th>\n",
       "      <td>local_file</td>\n",
       "      <td></td>\n",
       "      <td>nucl</td>\n",
       "      <td>SN.fq.gz</td>\n",
       "      <td>61504</td>\n",
       "      <td>supraneural_61504</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>supraneural</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>(0.978685122378, 0.507312601454, 0.456655135926)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPDI</th>\n",
       "      <td>local_file</td>\n",
       "      <td>distal_pars</td>\n",
       "      <td>nucl</td>\n",
       "      <td>SPDI.fq.gz</td>\n",
       "      <td>434JM</td>\n",
       "      <td>intest_distal_pars_434JM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPK</th>\n",
       "      <td>local_file</td>\n",
       "      <td>pars</td>\n",
       "      <td>nucl</td>\n",
       "      <td>SPK.fq.gz</td>\n",
       "      <td>434JM</td>\n",
       "      <td>kidney_pars_434JM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPPI</th>\n",
       "      <td>local_file</td>\n",
       "      <td>prox_pars</td>\n",
       "      <td>nucl</td>\n",
       "      <td>SPPI.fq.gz</td>\n",
       "      <td>434JM</td>\n",
       "      <td>intest_prox_pars_434JM</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SW2-Gill</th>\n",
       "      <td>local_file</td>\n",
       "      <td>saltw</td>\n",
       "      <td>nucl</td>\n",
       "      <td>SW2-Gill.fq.gz</td>\n",
       "      <td>614H7</td>\n",
       "      <td>gill_saltw_614H7</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gill</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>(0.978685122378, 0.507312601454, 0.456655135926)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SW2-Intestine-a</th>\n",
       "      <td>local_file</td>\n",
       "      <td>saltw_a</td>\n",
       "      <td>nucl</td>\n",
       "      <td>SW2-Intestine-a.fq.gz</td>\n",
       "      <td>614H7</td>\n",
       "      <td>intest_saltw_a_614H7</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SW2-Intestine-b</th>\n",
       "      <td>local_file</td>\n",
       "      <td>saltw_b</td>\n",
       "      <td>nucl</td>\n",
       "      <td>SW2-Intestine-b.fq.gz</td>\n",
       "      <td>614H7</td>\n",
       "      <td>intest_saltw_b_614H7</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SW2-Kidney</th>\n",
       "      <td>local_file</td>\n",
       "      <td>saltw</td>\n",
       "      <td>nucl</td>\n",
       "      <td>SW2-Kidney.fq.gz</td>\n",
       "      <td>614H7</td>\n",
       "      <td>kidney_saltw_614H7</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-1-Intestine</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg1</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-1-Intestine.fq.gz</td>\n",
       "      <td>614VD</td>\n",
       "      <td>intest_stg1_614VD</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-1-Kidney</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg1</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-1-Kidney.fq.gz</td>\n",
       "      <td>614VE</td>\n",
       "      <td>kidney_stg1_614VE</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-2-Intestine</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg2</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-2-Intestine.fq.gz</td>\n",
       "      <td>614VD</td>\n",
       "      <td>intest_stg2_614VD</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-2-Kidney</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg2</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-2-Kidney.fq.gz</td>\n",
       "      <td>614VE</td>\n",
       "      <td>kidney_stg2_614VE</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-3-Intestine</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg3</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-3-Intestine.fq.gz</td>\n",
       "      <td>614VD</td>\n",
       "      <td>intest_stg3_614VD</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-3-Kidney</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg3</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-3-Kidney.fq.gz</td>\n",
       "      <td>614VE</td>\n",
       "      <td>kidney_stg3_614VE</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-4-Intestine</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg4</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-4-Intestine.fq.gz</td>\n",
       "      <td>614VD</td>\n",
       "      <td>intest_stg4_614VD</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-4-Kidney</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg4</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-4-Kidney.fq.gz</td>\n",
       "      <td>614VE</td>\n",
       "      <td>kidney_stg4_614VE</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-5-Intestine</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg5</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-5-Intestine.fq.gz</td>\n",
       "      <td>614VD</td>\n",
       "      <td>intest_stg5_614VD</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-5-Kidney</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg5</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-5-Kidney.fq.gz</td>\n",
       "      <td>614VE</td>\n",
       "      <td>kidney_stg5_614VE</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-6-Intestine</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg6</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-6-Intestine.fq.gz</td>\n",
       "      <td>614VD</td>\n",
       "      <td>intest_stg6_614VD</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-6-Kidney</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg6</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-6-Kidney.fq.gz</td>\n",
       "      <td>614VE</td>\n",
       "      <td>kidney_stg6_614VE</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-7-Intestine</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg7</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-7-Intestine.fq.gz</td>\n",
       "      <td>614VD</td>\n",
       "      <td>intest_stg7_614VD</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intest</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>(0.991018839443, 0.706528275854, 0.384421383283)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stg-7-Kidney</th>\n",
       "      <td>local_file</td>\n",
       "      <td>stg7</td>\n",
       "      <td>nucl</td>\n",
       "      <td>Stg-7-Kidney.fq.gz</td>\n",
       "      <td>614VE</td>\n",
       "      <td>kidney_stg7_614VE</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>(0.98486735961, 0.804705893993, 0.892318345168)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>brain_paired</th>\n",
       "      <td>local_file</td>\n",
       "      <td></td>\n",
       "      <td>nucl</td>\n",
       "      <td>brain_paired.qc.fq.gz</td>\n",
       "      <td>30M2H</td>\n",
       "      <td>brain_30M2H</td>\n",
       "      <td>sample</td>\n",
       "      <td>True</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>brain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>(0.552941203117, 0.827450990677, 0.780392169952)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mssm_single_s1</th>\n",
       "      <td>local_file</td>\n",
       "      <td>ctrl</td>\n",
       "      <td>nucl</td>\n",
       "      <td>mssm_single_s1.trim.fq.gz</td>\n",
       "      <td>mssm</td>\n",
       "      <td>brain_ctrl_mssm</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>brain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>(0.552941203117, 0.827450990677, 0.780392169952)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mssm_single_s2</th>\n",
       "      <td>local_file</td>\n",
       "      <td>ctrl</td>\n",
       "      <td>nucl</td>\n",
       "      <td>mssm_single_s2.trim.fq.gz</td>\n",
       "      <td>mssm</td>\n",
       "      <td>spinal_ctrl_mssm</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>spinal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14</td>\n",
       "      <td>(0.749096516301, 0.733656301919, 0.852502892298)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mssm_single_s3</th>\n",
       "      <td>local_file</td>\n",
       "      <td>day3</td>\n",
       "      <td>nucl</td>\n",
       "      <td>mssm_single_s3.trim.fq.gz</td>\n",
       "      <td>mssm</td>\n",
       "      <td>brain_day3_mssm</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>brain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>(0.552941203117, 0.827450990677, 0.780392169952)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mssm_single_s5</th>\n",
       "      <td>local_file</td>\n",
       "      <td>day3</td>\n",
       "      <td>nucl</td>\n",
       "      <td>mssm_single_s5.trim.fq.gz</td>\n",
       "      <td>mssm</td>\n",
       "      <td>spinal_day3_mssm</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>spinal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14</td>\n",
       "      <td>(0.749096516301, 0.733656301919, 0.852502892298)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mssm_single_s6</th>\n",
       "      <td>local_file</td>\n",
       "      <td>day21</td>\n",
       "      <td>nucl</td>\n",
       "      <td>mssm_single_s6.trim.fq.gz</td>\n",
       "      <td>mssm</td>\n",
       "      <td>brain_day21_mssm</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>brain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>(0.552941203117, 0.827450990677, 0.780392169952)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mssm_single_s7</th>\n",
       "      <td>local_file</td>\n",
       "      <td>day21</td>\n",
       "      <td>nucl</td>\n",
       "      <td>mssm_single_s7.trim.fq.gz</td>\n",
       "      <td>mssm</td>\n",
       "      <td>spinal_day21_mssm</td>\n",
       "      <td>sample</td>\n",
       "      <td>False</td>\n",
       "      <td>phred64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+/-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>spinal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14</td>\n",
       "      <td>(0.749096516301, 0.733656301919, 0.852502892298)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       access        condition db_type  \\\n",
       "614GMAAXX_1_1_pf   local_file            stg18    nucl   \n",
       "614GMAAXX_2_1_pf   local_file            stg20    nucl   \n",
       "614GMAAXX_3_1_pf   local_file           stg22a    nucl   \n",
       "614GMAAXX_4_1_pf   local_file           stg22b    nucl   \n",
       "614GMAAXX_6_1_pf   local_file            stg23    nucl   \n",
       "614GMAAXX_7_1_pf   local_file          stg24c1    nucl   \n",
       "614GMAAXX_8_1_pf   local_file          stg24c2    nucl   \n",
       "AB057JABXX_s_7_pe  local_file           larval    nucl   \n",
       "AI                 local_file            adult    nucl   \n",
       "AK                 local_file            adult    nucl   \n",
       "BB0582ABXX_s_3_pe  local_file           larval    nucl   \n",
       "BB0582ABXX_s_4_pe  local_file             pars    nucl   \n",
       "BB0582ABXX_s_5_pe  local_file            adult    nucl   \n",
       "BB0582ABXX_s_6_pe  local_file          adult_1    nucl   \n",
       "BB0582ABXX_s_7_pe  local_file          adult_2    nucl   \n",
       "BB0582ABXX_s_8_pe  local_file          adult_3    nucl   \n",
       "FW1-Gill           local_file           freshw    nucl   \n",
       "FW1-Intestine      local_file           freshw    nucl   \n",
       "FW1-Kidney         local_file           freshw    nucl   \n",
       "G81-a              local_file    larval_a_tfm0    nucl   \n",
       "G81-b              local_file    larval_b_tfm0    nucl   \n",
       "G82-a              local_file    larval_a_tfm6    nucl   \n",
       "G82-b              local_file    larval_b_tfm6    nucl   \n",
       "G83-a              local_file   larval_a_tfm12    nucl   \n",
       "G83-b              local_file   larval_b_tfm12    nucl   \n",
       "Gill               local_file                     nucl   \n",
       "JV1_liver          local_file              juv    nucl   \n",
       "JV_intest          local_file            juv_b    nucl   \n",
       "JVa_liver          local_file            juv_1    nucl   \n",
       "JVb_liver          local_file            juv_a    nucl   \n",
       "Kidney             local_file                     nucl   \n",
       "L81-a              local_file    larval_a_tfm0    nucl   \n",
       "L81-b              local_file    larval_b_tfm0    nucl   \n",
       "L82-a              local_file    larval_a_tfm6    nucl   \n",
       "L82-b              local_file    larval_b_tfm6    nucl   \n",
       "L83-1-a            local_file  larval_a1_tfm12    nucl   \n",
       "L83-1-b            local_file  larval_b1_tfm12    nucl   \n",
       "L83-a              local_file   larval_a_tfm12    nucl   \n",
       "L83-b              local_file   larval_b_tfm12    nucl   \n",
       "LI                 local_file           larval    nucl   \n",
       "LK                 local_file           larval    nucl   \n",
       "Lips               local_file                     nucl   \n",
       "M0_liver           local_file           larval    nucl   \n",
       "M2_liver           local_file             stg2    nucl   \n",
       "M5_liver           local_file             stg5    nucl   \n",
       "Monocytes          local_file                     nucl   \n",
       "Muscle             local_file          m_sperm    nucl   \n",
       "Neutrophils        local_file                     nucl   \n",
       "OFHS               local_file            f_ovu    nucl   \n",
       "POF-eye            local_file         f_preovu    nucl   \n",
       "POF-ts             local_file         f_preovu    nucl   \n",
       "PSMGill            local_file       m_presperm    nucl   \n",
       "Rope               local_file          adult_m    nucl   \n",
       "SMGill             local_file          m_sperm    nucl   \n",
       "SMHS               local_file          m_sperm    nucl   \n",
       "SN                 local_file                     nucl   \n",
       "SPDI               local_file      distal_pars    nucl   \n",
       "SPK                local_file             pars    nucl   \n",
       "SPPI               local_file        prox_pars    nucl   \n",
       "SW2-Gill           local_file            saltw    nucl   \n",
       "SW2-Intestine-a    local_file          saltw_a    nucl   \n",
       "SW2-Intestine-b    local_file          saltw_b    nucl   \n",
       "SW2-Kidney         local_file            saltw    nucl   \n",
       "Stg-1-Intestine    local_file             stg1    nucl   \n",
       "Stg-1-Kidney       local_file             stg1    nucl   \n",
       "Stg-2-Intestine    local_file             stg2    nucl   \n",
       "Stg-2-Kidney       local_file             stg2    nucl   \n",
       "Stg-3-Intestine    local_file             stg3    nucl   \n",
       "Stg-3-Kidney       local_file             stg3    nucl   \n",
       "Stg-4-Intestine    local_file             stg4    nucl   \n",
       "Stg-4-Kidney       local_file             stg4    nucl   \n",
       "Stg-5-Intestine    local_file             stg5    nucl   \n",
       "Stg-5-Kidney       local_file             stg5    nucl   \n",
       "Stg-6-Intestine    local_file             stg6    nucl   \n",
       "Stg-6-Kidney       local_file             stg6    nucl   \n",
       "Stg-7-Intestine    local_file             stg7    nucl   \n",
       "Stg-7-Kidney       local_file             stg7    nucl   \n",
       "brain_paired       local_file                     nucl   \n",
       "mssm_single_s1     local_file             ctrl    nucl   \n",
       "mssm_single_s2     local_file             ctrl    nucl   \n",
       "mssm_single_s3     local_file             day3    nucl   \n",
       "mssm_single_s5     local_file             day3    nucl   \n",
       "mssm_single_s6     local_file            day21    nucl   \n",
       "mssm_single_s7     local_file            day21    nucl   \n",
       "\n",
       "                                       filename flowcell  \\\n",
       "614GMAAXX_1_1_pf      614GMAAXX_1_1_pf.qc.fq.gz    614GM   \n",
       "614GMAAXX_2_1_pf      614GMAAXX_2_1_pf.qc.fq.gz    614GM   \n",
       "614GMAAXX_3_1_pf      614GMAAXX_3_1_pf.qc.fq.gz    614GM   \n",
       "614GMAAXX_4_1_pf      614GMAAXX_4_1_pf.qc.fq.gz    614GM   \n",
       "614GMAAXX_6_1_pf      614GMAAXX_6_1_pf.qc.fq.gz    614GM   \n",
       "614GMAAXX_7_1_pf      614GMAAXX_7_1_pf.qc.fq.gz    614GM   \n",
       "614GMAAXX_8_1_pf      614GMAAXX_8_1_pf.qc.fq.gz    614GM   \n",
       "AB057JABXX_s_7_pe  AB057JABXX_s_7_pe.trim.fq.gz    AB057   \n",
       "AI                                     AI.fq.gz    614GM   \n",
       "AK                                     AK.fq.gz    614GM   \n",
       "BB0582ABXX_s_3_pe  BB0582ABXX_s_3_pe.trim.fq.gz    BB058   \n",
       "BB0582ABXX_s_4_pe  BB0582ABXX_s_4_pe.trim.fq.gz    BB058   \n",
       "BB0582ABXX_s_5_pe  BB0582ABXX_s_5_pe.trim.fq.gz    BB058   \n",
       "BB0582ABXX_s_6_pe  BB0582ABXX_s_6_pe.trim.fq.gz    BB058   \n",
       "BB0582ABXX_s_7_pe  BB0582ABXX_s_7_pe.trim.fq.gz    BB058   \n",
       "BB0582ABXX_s_8_pe  BB0582ABXX_s_8_pe.trim.fq.gz    BB058   \n",
       "FW1-Gill                         FW1-Gill.fq.gz    614H7   \n",
       "FW1-Intestine               FW1-Intestine.fq.gz    614H7   \n",
       "FW1-Kidney                     FW1-Kidney.fq.gz    614H7   \n",
       "G81-a                               G81-a.fq.gz    614GT   \n",
       "G81-b                               G81-b.fq.gz    61HWC   \n",
       "G82-a                               G82-a.fq.gz    614GT   \n",
       "G82-b                               G82-b.fq.gz    61HWC   \n",
       "G83-a                               G83-a.fq.gz    614GT   \n",
       "G83-b                               G83-b.fq.gz    61HWC   \n",
       "Gill                                 Gill.fq.gz    42UL4   \n",
       "JV1_liver                       JV1_liver.fq.gz    42UKJ   \n",
       "JV_intest                       JV_intest.fq.gz    42UKJ   \n",
       "JVa_liver                       JVa_liver.fq.gz    42UKJ   \n",
       "JVb_liver                       JVb_liver.fq.gz    42UKJ   \n",
       "Kidney                             Kidney.fq.gz    42UL4   \n",
       "L81-a                               L81-a.fq.gz    614GT   \n",
       "L81-b                               L81-b.fq.gz    61HWC   \n",
       "L82-a                               L82-a.fq.gz    614GT   \n",
       "L82-b                               L82-b.fq.gz    61HWC   \n",
       "L83-1-a                           L83-1-a.fq.gz    614GT   \n",
       "L83-1-b                           L83-1-b.fq.gz    61HWC   \n",
       "L83-a                               L83-a.fq.gz    614GT   \n",
       "L83-b                               L83-b.fq.gz    61HWC   \n",
       "LI                                     LI.fq.gz    434JM   \n",
       "LK                                     LK.fq.gz    434JM   \n",
       "Lips                                 Lips.fq.gz    42UL4   \n",
       "M0_liver                         M0_liver.fq.gz    42UKJ   \n",
       "M2_liver                         M2_liver.fq.gz    42UKJ   \n",
       "M5_liver                         M5_liver.fq.gz    42UKJ   \n",
       "Monocytes                       Monocytes.fq.gz    61504   \n",
       "Muscle                             Muscle.fq.gz    42UL4   \n",
       "Neutrophils                   Neutrophils.fq.gz    61504   \n",
       "OFHS                                 OFHS.fq.gz    61504   \n",
       "POF-eye                           POF-eye.fq.gz    61504   \n",
       "POF-ts                             POF-ts.fq.gz    61504   \n",
       "PSMGill                           PSMGill.fq.gz    42UL4   \n",
       "Rope                                 Rope.fq.gz    42UL4   \n",
       "SMGill                             SMGill.fq.gz    42UL4   \n",
       "SMHS                                 SMHS.fq.gz    61504   \n",
       "SN                                     SN.fq.gz    61504   \n",
       "SPDI                                 SPDI.fq.gz    434JM   \n",
       "SPK                                   SPK.fq.gz    434JM   \n",
       "SPPI                                 SPPI.fq.gz    434JM   \n",
       "SW2-Gill                         SW2-Gill.fq.gz    614H7   \n",
       "SW2-Intestine-a           SW2-Intestine-a.fq.gz    614H7   \n",
       "SW2-Intestine-b           SW2-Intestine-b.fq.gz    614H7   \n",
       "SW2-Kidney                     SW2-Kidney.fq.gz    614H7   \n",
       "Stg-1-Intestine           Stg-1-Intestine.fq.gz    614VD   \n",
       "Stg-1-Kidney                 Stg-1-Kidney.fq.gz    614VE   \n",
       "Stg-2-Intestine           Stg-2-Intestine.fq.gz    614VD   \n",
       "Stg-2-Kidney                 Stg-2-Kidney.fq.gz    614VE   \n",
       "Stg-3-Intestine           Stg-3-Intestine.fq.gz    614VD   \n",
       "Stg-3-Kidney                 Stg-3-Kidney.fq.gz    614VE   \n",
       "Stg-4-Intestine           Stg-4-Intestine.fq.gz    614VD   \n",
       "Stg-4-Kidney                 Stg-4-Kidney.fq.gz    614VE   \n",
       "Stg-5-Intestine           Stg-5-Intestine.fq.gz    614VD   \n",
       "Stg-5-Kidney                 Stg-5-Kidney.fq.gz    614VE   \n",
       "Stg-6-Intestine           Stg-6-Intestine.fq.gz    614VD   \n",
       "Stg-6-Kidney                 Stg-6-Kidney.fq.gz    614VE   \n",
       "Stg-7-Intestine           Stg-7-Intestine.fq.gz    614VD   \n",
       "Stg-7-Kidney                 Stg-7-Kidney.fq.gz    614VE   \n",
       "brain_paired              brain_paired.qc.fq.gz    30M2H   \n",
       "mssm_single_s1        mssm_single_s1.trim.fq.gz     mssm   \n",
       "mssm_single_s2        mssm_single_s2.trim.fq.gz     mssm   \n",
       "mssm_single_s3        mssm_single_s3.trim.fq.gz     mssm   \n",
       "mssm_single_s5        mssm_single_s5.trim.fq.gz     mssm   \n",
       "mssm_single_s6        mssm_single_s6.trim.fq.gz     mssm   \n",
       "mssm_single_s7        mssm_single_s7.trim.fq.gz     mssm   \n",
       "\n",
       "                                         label meta_type paired    phred  \\\n",
       "614GMAAXX_1_1_pf            embryo_stg18_614GM    sample  False  phred33   \n",
       "614GMAAXX_2_1_pf            embryo_stg20_614GM    sample  False  phred33   \n",
       "614GMAAXX_3_1_pf           embryo_stg22a_614GM    sample  False  phred33   \n",
       "614GMAAXX_4_1_pf           embryo_stg22b_614GM    sample  False  phred33   \n",
       "614GMAAXX_6_1_pf            embryo_stg23_614GM    sample  False  phred33   \n",
       "614GMAAXX_7_1_pf          embryo_stg24c1_614GM    sample  False  phred33   \n",
       "614GMAAXX_8_1_pf          embryo_stg24c2_614GM    sample  False  phred33   \n",
       "AB057JABXX_s_7_pe           brain_larval_AB057    sample   True  phred64   \n",
       "AI                          intest_adult_614GM    sample  False  phred33   \n",
       "AK                          kidney_adult_614GM    sample  False  phred33   \n",
       "BB0582ABXX_s_3_pe           liver_larval_BB058    sample   True  phred64   \n",
       "BB0582ABXX_s_4_pe             liver_pars_BB058    sample   True  phred64   \n",
       "BB0582ABXX_s_5_pe            brain_adult_BB058    sample   True  phred64   \n",
       "BB0582ABXX_s_6_pe      intestine_adult_1_BB058    sample   True  phred64   \n",
       "BB0582ABXX_s_7_pe      intestine_adult_2_BB058    sample   True  phred64   \n",
       "BB0582ABXX_s_8_pe        unknown_adult_3_BB058    sample   True  phred64   \n",
       "FW1-Gill                     gill_freshw_614H7    sample  False  phred33   \n",
       "FW1-Intestine              intest_freshw_614H7    sample  False  phred33   \n",
       "FW1-Kidney                 kidney_freshw_614H7    sample  False  phred33   \n",
       "G81-a                 gill_larval_a_tfm0_614GT    sample  False  phred33   \n",
       "G81-b                 gill_larval_b_tfm0_61HWC    sample  False  phred33   \n",
       "G82-a                 gill_larval_a_tfm6_614GT    sample  False  phred33   \n",
       "G82-b                 gill_larval_b_tfm6_61HWC    sample  False  phred33   \n",
       "G83-a                gill_larval_a_tfm12_614GT    sample  False  phred33   \n",
       "G83-b                gill_larval_b_tfm12_61HWC    sample  False  phred33   \n",
       "Gill                                gill_42UL4    sample  False  phred33   \n",
       "JV1_liver                     intest_juv_42UKJ    sample  False  phred33   \n",
       "JV_intest                    liver_juv_b_42UKJ    sample  False  phred33   \n",
       "JVa_liver                    liver_juv_1_42UKJ    sample  False  phred33   \n",
       "JVb_liver                    liver_juv_a_42UKJ    sample  False  phred33   \n",
       "Kidney                            kidney_42UL4    sample  False  phred33   \n",
       "L81-a                liver_larval_a_tfm0_614GT    sample  False  phred33   \n",
       "L81-b                liver_larval_b_tfm0_61HWC    sample  False  phred33   \n",
       "L82-a                liver_larval_a_tfm6_614GT    sample  False  phred33   \n",
       "L82-b                liver_larval_b_tfm6_61HWC    sample  False  phred33   \n",
       "L83-1-a            liver_larval_a1_tfm12_614GT    sample  False  phred33   \n",
       "L83-1-b            liver_larval_b1_tfm12_61HWC    sample  False  phred33   \n",
       "L83-a               liver_larval_a_tfm12_614GT    sample  False  phred33   \n",
       "L83-b               liver_larval_b_tfm12_61HWC    sample  False  phred33   \n",
       "LI                         intest_larval_434JM    sample  False  phred33   \n",
       "LK                         kidney_larval_434JM    sample  False  phred33   \n",
       "Lips                                lips_42UL4    sample  False  phred33   \n",
       "M0_liver                    liver_larval_42UKJ    sample  False  phred33   \n",
       "M2_liver                      liver_stg2_42UKJ    sample  False  phred33   \n",
       "M5_liver                      liver_stg5_42UKJ    sample  False  phred33   \n",
       "Monocytes                      monocytes_61504    sample  False  phred33   \n",
       "Muscle                    muscle_m_sperm_42UL4    sample  False  phred33   \n",
       "Neutrophils                  neutrophils_61504    sample  False  phred33   \n",
       "OFHS                      headskin_f_ovu_61504    sample  False  phred33   \n",
       "POF-eye                     eye_f_preovu_61504    sample  False  phred33   \n",
       "POF-ts                 tailskin_f_preovu_61504    sample  False  phred33   \n",
       "PSMGill                  gill_m_presperm_42UL4    sample  False  phred33   \n",
       "Rope                        rope_adult_m_42UL4    sample  False  phred33   \n",
       "SMGill                      gill_m_sperm_42UL4    sample  False  phred33   \n",
       "SMHS                    headskin_m_sperm_61504    sample  False  phred33   \n",
       "SN                           supraneural_61504    sample  False  phred33   \n",
       "SPDI                  intest_distal_pars_434JM    sample  False  phred33   \n",
       "SPK                          kidney_pars_434JM    sample  False  phred33   \n",
       "SPPI                    intest_prox_pars_434JM    sample  False  phred33   \n",
       "SW2-Gill                      gill_saltw_614H7    sample  False  phred33   \n",
       "SW2-Intestine-a           intest_saltw_a_614H7    sample  False  phred33   \n",
       "SW2-Intestine-b           intest_saltw_b_614H7    sample  False  phred33   \n",
       "SW2-Kidney                  kidney_saltw_614H7    sample  False  phred33   \n",
       "Stg-1-Intestine              intest_stg1_614VD    sample  False  phred33   \n",
       "Stg-1-Kidney                 kidney_stg1_614VE    sample  False  phred33   \n",
       "Stg-2-Intestine              intest_stg2_614VD    sample  False  phred33   \n",
       "Stg-2-Kidney                 kidney_stg2_614VE    sample  False  phred33   \n",
       "Stg-3-Intestine              intest_stg3_614VD    sample  False  phred33   \n",
       "Stg-3-Kidney                 kidney_stg3_614VE    sample  False  phred33   \n",
       "Stg-4-Intestine              intest_stg4_614VD    sample  False  phred33   \n",
       "Stg-4-Kidney                 kidney_stg4_614VE    sample  False  phred33   \n",
       "Stg-5-Intestine              intest_stg5_614VD    sample  False  phred33   \n",
       "Stg-5-Kidney                 kidney_stg5_614VE    sample  False  phred33   \n",
       "Stg-6-Intestine              intest_stg6_614VD    sample  False  phred33   \n",
       "Stg-6-Kidney                 kidney_stg6_614VE    sample  False  phred33   \n",
       "Stg-7-Intestine              intest_stg7_614VD    sample  False  phred33   \n",
       "Stg-7-Kidney                 kidney_stg7_614VE    sample  False  phred33   \n",
       "brain_paired                       brain_30M2H    sample   True  phred64   \n",
       "mssm_single_s1                 brain_ctrl_mssm    sample  False  phred33   \n",
       "mssm_single_s2                spinal_ctrl_mssm    sample  False  phred64   \n",
       "mssm_single_s3                 brain_day3_mssm    sample  False  phred64   \n",
       "mssm_single_s5                spinal_day3_mssm    sample  False  phred64   \n",
       "mssm_single_s6                brain_day21_mssm    sample  False  phred64   \n",
       "mssm_single_s7               spinal_day21_mssm    sample  False  phred64   \n",
       "\n",
       "                  q_type strand terms       tissue  url  tissue_c  \\\n",
       "614GMAAXX_1_1_pf     NaN    +/-   NaN       embryo  NaN         1   \n",
       "614GMAAXX_2_1_pf     NaN    +/-   NaN       embryo  NaN         1   \n",
       "614GMAAXX_3_1_pf     NaN    +/-   NaN       embryo  NaN         1   \n",
       "614GMAAXX_4_1_pf     NaN    +/-   NaN       embryo  NaN         1   \n",
       "614GMAAXX_6_1_pf     NaN    +/-   NaN       embryo  NaN         1   \n",
       "614GMAAXX_7_1_pf     NaN    +/-   NaN       embryo  NaN         1   \n",
       "614GMAAXX_8_1_pf     NaN    +/-   NaN       embryo  NaN         1   \n",
       "AB057JABXX_s_7_pe    NaN    +/-   NaN        brain  NaN         0   \n",
       "AI                   NaN    +/-   NaN       intest  NaN         5   \n",
       "AK                   NaN    +/-   NaN       kidney  NaN         7   \n",
       "BB0582ABXX_s_3_pe    NaN    +/-   NaN        liver  NaN         9   \n",
       "BB0582ABXX_s_4_pe    NaN    +/-   NaN        liver  NaN         9   \n",
       "BB0582ABXX_s_5_pe    NaN    +/-   NaN        brain  NaN         0   \n",
       "BB0582ABXX_s_6_pe    NaN    +/-   NaN    intestine  NaN         6   \n",
       "BB0582ABXX_s_7_pe    NaN    +/-   NaN    intestine  NaN         6   \n",
       "BB0582ABXX_s_8_pe    NaN    +/-   NaN      unknown  NaN        17   \n",
       "FW1-Gill             NaN    +/-   NaN         gill  NaN         3   \n",
       "FW1-Intestine        NaN    +/-   NaN       intest  NaN         5   \n",
       "FW1-Kidney           NaN    +/-   NaN       kidney  NaN         7   \n",
       "G81-a                NaN    +/-   NaN         gill  NaN         3   \n",
       "G81-b                NaN    +/-   NaN         gill  NaN         3   \n",
       "G82-a                NaN    +/-   NaN         gill  NaN         3   \n",
       "G82-b                NaN    +/-   NaN         gill  NaN         3   \n",
       "G83-a                NaN    +/-   NaN         gill  NaN         3   \n",
       "G83-b                NaN    +/-   NaN         gill  NaN         3   \n",
       "Gill                 NaN    +/-   NaN         gill  NaN         3   \n",
       "JV1_liver            NaN    +/-   NaN       intest  NaN         5   \n",
       "JV_intest            NaN    +/-   NaN        liver  NaN         9   \n",
       "JVa_liver            NaN    +/-   NaN        liver  NaN         9   \n",
       "JVb_liver            NaN    +/-   NaN        liver  NaN         9   \n",
       "Kidney               NaN    +/-   NaN       kidney  NaN         7   \n",
       "L81-a                NaN    +/-   NaN        liver  NaN         9   \n",
       "L81-b                NaN    +/-   NaN        liver  NaN         9   \n",
       "L82-a                NaN    +/-   NaN        liver  NaN         9   \n",
       "L82-b                NaN    +/-   NaN        liver  NaN         9   \n",
       "L83-1-a              NaN    +/-   NaN        liver  NaN         9   \n",
       "L83-1-b              NaN    +/-   NaN        liver  NaN         9   \n",
       "L83-a                NaN    +/-   NaN        liver  NaN         9   \n",
       "L83-b                NaN    +/-   NaN        liver  NaN         9   \n",
       "LI                   NaN    +/-   NaN       intest  NaN         5   \n",
       "LK                   NaN    +/-   NaN       kidney  NaN         7   \n",
       "Lips                 NaN    +/-   NaN         lips  NaN         8   \n",
       "M0_liver             NaN    +/-   NaN        liver  NaN         9   \n",
       "M2_liver             NaN    +/-   NaN        liver  NaN         9   \n",
       "M5_liver             NaN    +/-   NaN        liver  NaN         9   \n",
       "Monocytes            NaN    +/-   NaN    monocytes  NaN        10   \n",
       "Muscle               NaN    +/-   NaN       muscle  NaN        11   \n",
       "Neutrophils          NaN    +/-   NaN  neutrophils  NaN        12   \n",
       "OFHS                 NaN    +/-   NaN     headskin  NaN         4   \n",
       "POF-eye              NaN    +/-   NaN          eye  NaN         2   \n",
       "POF-ts               NaN    +/-   NaN     tailskin  NaN        16   \n",
       "PSMGill              NaN    +/-   NaN         gill  NaN         3   \n",
       "Rope                 NaN    +/-   NaN         rope  NaN        13   \n",
       "SMGill               NaN    +/-   NaN         gill  NaN         3   \n",
       "SMHS                 NaN    +/-   NaN     headskin  NaN         4   \n",
       "SN                   NaN    +/-   NaN  supraneural  NaN        15   \n",
       "SPDI                 NaN    +/-   NaN       intest  NaN         5   \n",
       "SPK                  NaN    +/-   NaN       kidney  NaN         7   \n",
       "SPPI                 NaN    +/-   NaN       intest  NaN         5   \n",
       "SW2-Gill             NaN    +/-   NaN         gill  NaN         3   \n",
       "SW2-Intestine-a      NaN    +/-   NaN       intest  NaN         5   \n",
       "SW2-Intestine-b      NaN    +/-   NaN       intest  NaN         5   \n",
       "SW2-Kidney           NaN    +/-   NaN       kidney  NaN         7   \n",
       "Stg-1-Intestine      NaN    +/-   NaN       intest  NaN         5   \n",
       "Stg-1-Kidney         NaN    +/-   NaN       kidney  NaN         7   \n",
       "Stg-2-Intestine      NaN    +/-   NaN       intest  NaN         5   \n",
       "Stg-2-Kidney         NaN    +/-   NaN       kidney  NaN         7   \n",
       "Stg-3-Intestine      NaN    +/-   NaN       intest  NaN         5   \n",
       "Stg-3-Kidney         NaN    +/-   NaN       kidney  NaN         7   \n",
       "Stg-4-Intestine      NaN    +/-   NaN       intest  NaN         5   \n",
       "Stg-4-Kidney         NaN    +/-   NaN       kidney  NaN         7   \n",
       "Stg-5-Intestine      NaN    +/-   NaN       intest  NaN         5   \n",
       "Stg-5-Kidney         NaN    +/-   NaN       kidney  NaN         7   \n",
       "Stg-6-Intestine      NaN    +/-   NaN       intest  NaN         5   \n",
       "Stg-6-Kidney         NaN    +/-   NaN       kidney  NaN         7   \n",
       "Stg-7-Intestine      NaN    +/-   NaN       intest  NaN         5   \n",
       "Stg-7-Kidney         NaN    +/-   NaN       kidney  NaN         7   \n",
       "brain_paired         NaN    +/-   NaN        brain  NaN         0   \n",
       "mssm_single_s1       NaN    +/-   NaN        brain  NaN         0   \n",
       "mssm_single_s2       NaN    +/-   NaN       spinal  NaN        14   \n",
       "mssm_single_s3       NaN    +/-   NaN        brain  NaN         0   \n",
       "mssm_single_s5       NaN    +/-   NaN       spinal  NaN        14   \n",
       "mssm_single_s6       NaN    +/-   NaN        brain  NaN         0   \n",
       "mssm_single_s7       NaN    +/-   NaN       spinal  NaN        14   \n",
       "\n",
       "                                                              color  \n",
       "614GMAAXX_1_1_pf   (0.996493656495, 0.998646674437, 0.702575950062)  \n",
       "614GMAAXX_2_1_pf   (0.996493656495, 0.998646674437, 0.702575950062)  \n",
       "614GMAAXX_3_1_pf   (0.996493656495, 0.998646674437, 0.702575950062)  \n",
       "614GMAAXX_4_1_pf   (0.996493656495, 0.998646674437, 0.702575950062)  \n",
       "614GMAAXX_6_1_pf   (0.996493656495, 0.998646674437, 0.702575950062)  \n",
       "614GMAAXX_7_1_pf   (0.996493656495, 0.998646674437, 0.702575950062)  \n",
       "614GMAAXX_8_1_pf   (0.996493656495, 0.998646674437, 0.702575950062)  \n",
       "AB057JABXX_s_7_pe  (0.552941203117, 0.827450990677, 0.780392169952)  \n",
       "AI                 (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "AK                  (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "BB0582ABXX_s_3_pe  (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "BB0582ABXX_s_4_pe  (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "BB0582ABXX_s_5_pe  (0.552941203117, 0.827450990677, 0.780392169952)  \n",
       "BB0582ABXX_s_6_pe  (0.703098825847, 0.869942337625, 0.411657060245)  \n",
       "BB0582ABXX_s_7_pe  (0.703098825847, 0.869942337625, 0.411657060245)  \n",
       "BB0582ABXX_s_8_pe  (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "FW1-Gill           (0.978685122378, 0.507312601454, 0.456655135926)  \n",
       "FW1-Intestine      (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "FW1-Kidney          (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "G81-a              (0.978685122378, 0.507312601454, 0.456655135926)  \n",
       "G81-b              (0.978685122378, 0.507312601454, 0.456655135926)  \n",
       "G82-a              (0.978685122378, 0.507312601454, 0.456655135926)  \n",
       "G82-b              (0.978685122378, 0.507312601454, 0.456655135926)  \n",
       "G83-a              (0.978685122378, 0.507312601454, 0.456655135926)  \n",
       "G83-b              (0.978685122378, 0.507312601454, 0.456655135926)  \n",
       "Gill               (0.978685122378, 0.507312601454, 0.456655135926)  \n",
       "JV1_liver          (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "JV_intest          (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "JVa_liver          (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "JVb_liver          (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "Kidney              (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "L81-a              (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "L81-b              (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "L82-a              (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "L82-b              (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "L83-1-a            (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "L83-1-b            (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "L83-a              (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "L83-b              (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "LI                 (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "LK                  (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "Lips               (0.848304507312, 0.842768175462, 0.848396779509)  \n",
       "M0_liver           (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "M2_liver           (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "M5_liver           (0.738239154395, 0.508542897421, 0.741668604402)  \n",
       "Monocytes          (0.801568639278, 0.921630146924, 0.769903896603)  \n",
       "Muscle                        (1.0, 0.929411768913, 0.435294121504)  \n",
       "Neutrophils        (0.552941203117, 0.827450990677, 0.780392169952)  \n",
       "OFHS               (0.507727826343, 0.694256073587, 0.822237611168)  \n",
       "POF-eye            (0.749096516301, 0.733656301919, 0.852502892298)  \n",
       "POF-ts             (0.507727826343, 0.694256073587, 0.822237611168)  \n",
       "PSMGill            (0.978685122378, 0.507312601454, 0.456655135926)  \n",
       "Rope               (0.996493656495, 0.998646674437, 0.702575950062)  \n",
       "SMGill             (0.978685122378, 0.507312601454, 0.456655135926)  \n",
       "SMHS               (0.507727826343, 0.694256073587, 0.822237611168)  \n",
       "SN                 (0.978685122378, 0.507312601454, 0.456655135926)  \n",
       "SPDI               (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "SPK                 (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "SPPI               (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "SW2-Gill           (0.978685122378, 0.507312601454, 0.456655135926)  \n",
       "SW2-Intestine-a    (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "SW2-Intestine-b    (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "SW2-Kidney          (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "Stg-1-Intestine    (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "Stg-1-Kidney        (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "Stg-2-Intestine    (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "Stg-2-Kidney        (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "Stg-3-Intestine    (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "Stg-3-Kidney        (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "Stg-4-Intestine    (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "Stg-4-Kidney        (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "Stg-5-Intestine    (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "Stg-5-Kidney        (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "Stg-6-Intestine    (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "Stg-6-Kidney        (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "Stg-7-Intestine    (0.991018839443, 0.706528275854, 0.384421383283)  \n",
       "Stg-7-Kidney        (0.98486735961, 0.804705893993, 0.892318345168)  \n",
       "brain_paired       (0.552941203117, 0.827450990677, 0.780392169952)  \n",
       "mssm_single_s1     (0.552941203117, 0.827450990677, 0.780392169952)  \n",
       "mssm_single_s2     (0.749096516301, 0.733656301919, 0.852502892298)  \n",
       "mssm_single_s3     (0.552941203117, 0.827450990677, 0.780392169952)  \n",
       "mssm_single_s5     (0.749096516301, 0.733656301919, 0.852502892298)  \n",
       "mssm_single_s6     (0.552941203117, 0.827450990677, 0.780392169952)  \n",
       "mssm_single_s7     (0.749096516301, 0.733656301919, 0.852502892298)  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create an HDF5 volume to store data needed in the other notebooks; we won't be performing ops directly on it, so use maximum compression to save disk space. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "store = pd.HDFStore(wdir('{}.store.h5'.format(prefix)), complib='zlib', complevel=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<function __main__.exit_func>"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import atexit\n",
    "def exit_func():\n",
    "    dump_results()\n",
    "    store.close()\n",
    "atexit.register(exit_func)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "for fn in resources_df[resources_df.meta_type == 'assembly'].filename:\n",
    "    screed.read_fasta_sequences(wdir(fn))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Transcript Support"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "tpm_df = pd.read_csv(wdir('lamp10.eXpress.tpm.tsv'), delimiter='\\t', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "labels = dict(zip(sample_df.filename, sample_df.label))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "tpm_df.rename(columns=labels, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "tpm_df.sort(axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/tables/path.py:100: NaturalNameWarning: object name is not a valid Python identifier: 'lamp10.eXpress.tpm.tsv'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  NaturalNameWarning)\n",
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/pandas/io/pytables.py:2577: PerformanceWarning: \n",
      "your performance may suffer as PyTables will pickle object types that it cannot\n",
      "map directly to c-types [inferred_type->unicode,key->axis0] [items->None]\n",
      "\n",
      "  warnings.warn(ws, PerformanceWarning)\n",
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/pandas/io/pytables.py:2577: PerformanceWarning: \n",
      "your performance may suffer as PyTables will pickle object types that it cannot\n",
      "map directly to c-types [inferred_type->unicode,key->block0_items] [items->None]\n",
      "\n",
      "  warnings.warn(ws, PerformanceWarning)\n"
     ]
    }
   ],
   "source": [
    "store['lamp10.eXpress.tpm.tsv'] = tpm_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Blast Results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%load_ext cython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%cython\n",
    "cimport numpy as np\n",
    "import numpy as np\n",
    "# Returns: \n",
    "# [0: sstart\n",
    "#  1: send\n",
    "#  2: qstart\n",
    "#  3: qend\n",
    "#  4: sstrand\n",
    "#  5: qstrand]\n",
    "cdef np.ndarray[long] fix_coords_single(long sstart, long send, long qstart, long qend):\n",
    "    cdef np.ndarray[long] res = np.empty(6, dtype=long)\n",
    "    \n",
    "    if sstart < send:\n",
    "        res[0] = sstart - 1\n",
    "        res[1] = send\n",
    "        res[4] = 1\n",
    "    else:\n",
    "        res[0] = send\n",
    "        res[1] = sstart + 1\n",
    "        res[4] = -1\n",
    "    \n",
    "    if qstart < qend:\n",
    "        res[2] = qstart - 1\n",
    "        res[3] = qend\n",
    "        res[5] = 1\n",
    "    else:\n",
    "        res[2] = qend\n",
    "        res[3] = qstart + 1\n",
    "        res[5] = -1\n",
    "    \n",
    "    return res\n",
    "\n",
    "cpdef np.ndarray[long, ndim=2] fix_blast_coords(np.ndarray[long] sstart, np.ndarray[long] send, \n",
    "                                                    np.ndarray[long] qstart, np.ndarray[long] qend):\n",
    "    cdef long n = len(sstart)\n",
    "    cdef long i = 0\n",
    "    cdef np.ndarray[long, ndim=2] res = np.empty((n,6), dtype=long)\n",
    "    for i in range(n):\n",
    "        res[i,:] = fix_coords_single(sstart[i], send[i], qstart[i], qend[i])\n",
    "    return res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import blasttools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def fix_blast_coords_df(df):\n",
    "    coords = fix_blast_coords(df.sstart.values, df.send.values, df.qstart.values, df.qend.values)\n",
    "    df['sstart'] = coords[:,0]\n",
    "    df['send'] = coords[:,1]\n",
    "    df['qstart'] = coords[:,2]\n",
    "    df['qend'] = coords[:,3]\n",
    "    df['sstrand'] = coords[:,4]\n",
    "    df['qstrand'] = coords[:,5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Myxinidae_pep lamp10.fasta.x.Myx.pep.all.fa.db.tsv\n",
      "braFlo_pep"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/tables/path.py:100: NaturalNameWarning: object name is not a valid Python identifier: 'lamp10.fasta.x.Myx.pep.all.fa.db.tsv'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  NaturalNameWarning)\n",
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/tables/path.py:100: NaturalNameWarning: object name is not a valid Python identifier: 'lamp10.fasta.x.braFlo.pep.all.fa.db.tsv'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  NaturalNameWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " lamp10.fasta.x.braFlo.pep.all.fa.db.tsv\n",
      "danRer_pep lamp10.fasta.x.danRer.pep.fa.db.tsv\n",
      "homSap_pep"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/tables/path.py:100: NaturalNameWarning: object name is not a valid Python identifier: 'lamp10.fasta.x.danRer.pep.fa.db.tsv'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  NaturalNameWarning)\n",
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/tables/path.py:100: NaturalNameWarning: object name is not a valid Python identifier: 'lamp10.fasta.x.homSap.pep.fa.db.tsv'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  NaturalNameWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " lamp10.fasta.x.homSap.pep.fa.db.tsv\n",
      "musMus_pep lamp10.fasta.x.musMus.pep.fa.db.tsv\n",
      "petMar2"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/tables/path.py:100: NaturalNameWarning: object name is not a valid Python identifier: 'lamp10.fasta.x.musMus.pep.fa.db.tsv'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  NaturalNameWarning)\n",
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/tables/path.py:100: NaturalNameWarning: object name is not a valid Python identifier: 'lamp10.fasta.x.petMar2.fa.db.tsv'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  NaturalNameWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " lamp10.fasta.x.petMar2.fa.db.tsv\n",
      "petMar2_cdna lamp10.fasta.x.petMar2.cdna.fa.db.tsv\n",
      "petMar2_cds"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/tables/path.py:100: NaturalNameWarning: object name is not a valid Python identifier: 'lamp10.fasta.x.petMar2.cdna.fa.db.tsv'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  NaturalNameWarning)\n",
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/tables/path.py:100: NaturalNameWarning: object name is not a valid Python identifier: 'lamp10.fasta.x.petMar2.cds.fa.db.tsv'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  NaturalNameWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " lamp10.fasta.x.petMar2.cds.fa.db.tsv\n",
      "petMar2_ncrna lamp10.fasta.x.petMar2.ncrna.fa.db.tsv\n",
      "petMar2_pep"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/tables/path.py:100: NaturalNameWarning: object name is not a valid Python identifier: 'lamp10.fasta.x.petMar2.ncrna.fa.db.tsv'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  NaturalNameWarning)\n",
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/tables/path.py:100: NaturalNameWarning: object name is not a valid Python identifier: 'lamp10.fasta.x.petMar2.pep.fa.db.tsv'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  NaturalNameWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " lamp10.fasta.x.petMar2.pep.fa.db.tsv\n"
     ]
    }
   ],
   "source": [
    "blast_items = resources_df[resources_df.meta_type.isin(['sample', 'assembly', 'gtf_database']) == False]\n",
    "for i, (dbname, info) in enumerate(blast_items.iterrows()):\n",
    "    target = '{}.fasta.x.{}.db.tsv'.format('lamp10', info['filename'])\n",
    "    print dbname, target\n",
    "    \n",
    "    df = blasttools.blast_to_df(wdir(target))\n",
    "    fix_blast_coords_df(df)\n",
    "\n",
    "    store[target] = df\n",
    "\n",
    "    tmp = pd.merge(pd.DataFrame(index=tpm_df.index), df,\n",
    "                      left_index=True, right_index=True, how='left')\n",
    "    blasttools.best_hits(tmp)\n",
    "\n",
    "    if i == 0:\n",
    "        lamp10_best_hits = pd.Panel({target: tmp})\n",
    "    else:\n",
    "        lamp10_best_hits[target] = tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/pandas/io/pytables.py:2577: PerformanceWarning: \n",
      "your performance may suffer as PyTables will pickle object types that it cannot\n",
      "map directly to c-types [inferred_type->mixed,key->block0_values] [items->['lamp10.fasta.x.Myx.pep.all.fa.db.tsv', 'lamp10.fasta.x.braFlo.pep.all.fa.db.tsv', 'lamp10.fasta.x.danRer.pep.fa.db.tsv', 'lamp10.fasta.x.homSap.pep.fa.db.tsv', 'lamp10.fasta.x.musMus.pep.fa.db.tsv', 'lamp10.fasta.x.petMar2.fa.db.tsv', 'lamp10.fasta.x.petMar2.cdna.fa.db.tsv', 'lamp10.fasta.x.petMar2.cds.fa.db.tsv', 'lamp10.fasta.x.petMar2.ncrna.fa.db.tsv', 'lamp10.fasta.x.petMar2.pep.fa.db.tsv']]\n",
      "\n",
      "  warnings.warn(ws, PerformanceWarning)\n"
     ]
    }
   ],
   "source": [
    "store['lamp10_best_hits'] = lamp10_best_hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "lamp10.fasta.x.Myx.pep.all.fa.db.tsv <=> Myx.pep.all.fa.db.x.lamp10.fasta.tsv\n",
      "lamp10.fasta.x.braFlo.pep.all.fa.db.tsv <=> braFlo.pep.all.fa.db.x.lamp10.fasta.tsv\n",
      "lamp10.fasta.x.danRer.pep.fa.db.tsv <=> danRer.pep.fa.db.x.lamp10.fasta.tsv\n",
      "lamp10.fasta.x.homSap.pep.fa.db.tsv <=> homSap.pep.fa.db.x.lamp10.fasta.tsv\n",
      "lamp10.fasta.x.musMus.pep.fa.db.tsv <=> musMus.pep.fa.db.x.lamp10.fasta.tsv\n",
      "lamp10.fasta.x.petMar2.fa.db.tsv <=> petMar2.fa.db.x.lamp10.fasta.tsv\n",
      "lamp10.fasta.x.petMar2.cdna.fa.db.tsv <=> petMar2.cdna.fa.db.x.lamp10.fasta.tsv\n",
      "lamp10.fasta.x.petMar2.cds.fa.db.tsv <=> petMar2.cds.fa.db.x.lamp10.fasta.tsv\n",
      "lamp10.fasta.x.petMar2.ncrna.fa.db.tsv <=> petMar2.ncrna.fa.db.x.lamp10.fasta.tsv\n",
      "lamp10.fasta.x.petMar2.pep.fa.db.tsv <=> petMar2.pep.fa.db.x.lamp10.fasta.tsv\n"
     ]
    }
   ],
   "source": [
    "blast_items = resources_df[resources_df.meta_type.isin(['sample', 'assembly', 'gtf_database']) == False]\n",
    "for i, (dbname, info) in enumerate(blast_items.iterrows()):\n",
    "    A_fn = '{}.fasta.x.{}.db.tsv'.format('lamp10', info.filename)\n",
    "    B_fn = '{}.db.x.{}.fasta.tsv'.format(info.filename, 'lamp10')\n",
    "    \n",
    "    print '{} <=> {}'.format(A_fn, B_fn)\n",
    "    \n",
    "    A = pd.read_table(wdir(A_fn), header=None, index_col=0, names=outfmt6)\n",
    "    B = pd.read_table(wdir(B_fn), header=None, index_col=0, names=outfmt6)\n",
    "\n",
    "    fix_blast_coords_df(A)\n",
    "    fix_blast_coords_df(B)\n",
    "    \n",
    "    X = blasttools.get_orthologies(A, B, tpm_df.index)\n",
    "    \n",
    "    if i == 0:\n",
    "        lamp10_ortho = pd.Panel({A_fn: X})\n",
    "    else:\n",
    "        lamp10_ortho[A_fn] = X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "store['lamp10_ortho'] = lamp10_ortho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[u'../_work/petMar2.cdna.fa.x.Myx.pep.all.fa.db.tsv',\n",
       " u'../_work/petMar2.cdna.fa.x.homSap.pep.fa.db.tsv',\n",
       " u'../_work/petMar2.cdna.fa.x.petMar2.pep.fa.db.tsv',\n",
       " u'../_work/petMar2.cdna.fa.x.danRer.pep.fa.db.tsv',\n",
       " u'../_work/petMar2.cdna.fa.x.musMus.pep.fa.db.tsv',\n",
       " u'../_work/petMar2.cdna.fa.x.petMar2.fa.db.tsv',\n",
       " u'../_work/petMar2.cdna.fa.x.braFlo.pep.all.fa.db.tsv']"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "glob.glob(wdir('petMar2.cdna.fa.x*.tsv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/camille/miniconda/envs/bio/lib/python2.7/site-packages/tables/path.py:100: NaturalNameWarning: object name is not a valid Python identifier: 'petMar2.cdna.fa.x.petMar2.fa.db.tsv'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  NaturalNameWarning)\n"
     ]
    }
   ],
   "source": [
    "petMar2_cdna_x_petMar2 = blasttools.blast_to_df(wdir('petMar2.cdna.fa.x.petMar2.fa.db.tsv'))\n",
    "fix_blast_coords_df(petMar2_cdna_x_petMar2)\n",
    "store['petMar2.cdna.fa.x.petMar2.fa.db.tsv'] = petMar2_cdna_x_petMar2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "lamp10_blast_filter_df = lamp10_best_hits.minor_xs('evalue') >= 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "lamp10_ortho_filter_df = lamp10_ortho.minor_xs('evalue_x') >= 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "store['lamp10_blast_filter_df'] = lamp10_blast_filter_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "store['lamp10_ortho_filter_df'] = lamp10_ortho_filter_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "tissue_tr_df = (tpm_df > 0).groupby(by=sample_df.sort(columns='label').tissue.values, axis=1).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "store['tissue_tr_df'] = tissue_tr_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "store.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
